src/test/java/de/konfidas/ttc/messages/logtime/GeneralizedLogTimeTests.java

```diff
@@ -0,0 +1,129 @@
+package de.konfidas.ttc.messages.logtime;
+
+import org.bouncycastle.asn1.ASN1GeneralizedTime;
+import org.junit.Test;
+
+import java.text.ParseException;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class GeneralizedLogTimeTests {
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Test equal time stamps:
+
+    @Test
+    public void testEqualGeneralizedTime() throws ParseException {
+
+        ASN1GeneralizedTime element = new ASN1GeneralizedTime("20200421124355Z");
+
+        GeneralizedLogTime t1 = new GeneralizedLogTime(element);
+        GeneralizedLogTime t2 = new GeneralizedLogTime(element);
+
+        assertTrue(t1.wasNotAfter(t2));
+        assertTrue(t2.wasNotAfter(t1));
+    }
+
+    @Test
+    public void testEqualGeneralizedTime2() throws ParseException {
+        GeneralizedLogTime t1 = new GeneralizedLogTime(new ASN1GeneralizedTime("20200421124355Z"));
+        GeneralizedLogTime t2 = new GeneralizedLogTime(new ASN1GeneralizedTime("20200421124355+0000"));
+
+        assertTrue(t1.wasNotAfter(t2));
+        assertTrue(t2.wasNotAfter(t1));
+    }
+
+    @Test
+    public void testEqualGeneralizedTime3() throws ParseException {
+        GeneralizedLogTime t1 = new GeneralizedLogTime(new ASN1GeneralizedTime("20200421124355Z"));
+        GeneralizedLogTime t2 = new GeneralizedLogTime(new ASN1GeneralizedTime("20200421124355-0000"));
+
+        assertTrue(t1.wasNotAfter(t2));
+        assertTrue(t2.wasNotAfter(t1));
+    }
+
+    @Test
+    public void testEqualGeneralizedTime4() throws ParseException {
+        GeneralizedLogTime t1 = new GeneralizedLogTime(new ASN1GeneralizedTime("20200421124355Z"));
+        GeneralizedLogTime t2 = new GeneralizedLogTime(new ASN1GeneralizedTime("20200421124455+0001"));
+
+        assertTrue(t1.wasNotAfter(t2));
+        assertTrue(t2.wasNotAfter(t1));
+    }
+
+    @Test
+    public void testEqualGeneralizedTime4b() throws ParseException {
+        GeneralizedLogTime t1 = new GeneralizedLogTime(new ASN1GeneralizedTime("20200421125955Z"));
+        GeneralizedLogTime t2 = new GeneralizedLogTime(new ASN1GeneralizedTime("20200421130055+0001"));

```

Comment:
Diesen FAll verstehe ich nicht ganz. Warum sind die Zeiten noch gleich?

Reply:
For Readability: `2020-04-21-12-59-55Z` and `2020-04-21-13-00-55+0001` 


`2020-04-21-12-59-55Z` and `2020-04-21-13-00-55Z` are off by one minute (YYYY-DD-MM-hh-mm-ss), the `+0001` uses the timezone encoding possibility of `GeneralizedTime `to fix this one minute off.

Note that I use **BouncyCastle**-Classes to parse the data into a Java `Date `instanz and I compare the content of the `Dates`. So the hard part of the work is done elsewhere...